prody/proteins/interactions.py

```diff
@@ -3786,3 +4047,192 @@ def parseLigandInteractions(self, filename):
+        :type selection: str
+
+        :arg contacts_min: Minimal number of contacts which residue may form with ligand.
+                            by default is 2. 
+        :type contacts_min: int  """
+
+        atoms = self._atoms   
+        interactions = self._interactions_traj
+        contacts_min = kwargs.pop('contacts_min', 2)
+        selection = kwargs.pop('selection', None)
+        
+        from collections import Counter
+
+        if selection == 'total':  # Compute all interactions without distinguishing ligands
+            all_residues = [ j[1]+j[3] for i in interactions for j in i ]
+            dictOfInteractions = Counter(all_residues)
+        
+        else:
+            interactions2 = [element for group in interactions for element in group]
+            ligs = {}
+            dictOfInteractions = []
+            for i in interactions2:
+                ligs_names = i[4]+i[6]
+                if ligs_names not in ligs:
+                    ligs[ligs_names] = []
+                    
+                res_name = i[1]+i[3]
+                ligs[ligs_names].append(res_name)
+
+            for i in ligs.keys():
+
+                if selection == None:
+                    LOGGER.info('LIGAND: {0}'.format(i))
+                    aa_counter = Counter(ligs[i])
+                    dictOfInteractions.append(aa_counter)
+                    
+                else:
+                    if selection not in ligs.keys():
+                        LOGGER.info('Wrong selection. Please provide ligand name with chain ID.')
+                    if i == selection:
+                        LOGGER.info('LIGAND: {0}'.format(selection))
+                        aa_counter = Counter(ligs[selection])
+                        dictOfInteractions.append(aa_counter)                    
+                
+        self._freq_interactors = dictOfInteractions
+        
+        return dictOfInteractions
+
+
+    def saveInteractionsPDB(self, **kwargs):
+        """Save the number of interactions with ligand to PDB file in occupancy column
+        It will recognize the chains. If the system will contain one chain and many segments
+        the PDB file will not be created in a correct way.
+        
+        :arg filename: name of the PDB file which will be saved for visualization,
+                     it will contain the results in occupancy column.
+        :type filename: str  
+        
+        :arg ligand_sele: ligand selection,
+                          by default is 'all not (protein or water or ion)'.
+        :type ligand_sele: str          
+        """
+        
+        if self._freq_interactors is None:
+            raise ValueError('Please calculate frequent interactors using getFrequentInteractors.')
+
+        atoms = self._atoms     
+        dictOfInteractions = self._freq_interactors
+        ligand_sele = kwargs.pop('ligand_sele', 'all not (protein or water or ion)')
+
+        chids_list = np.unique(atoms.getChids())
```

Comment:
perhaps we could use getHierView instead to get both segments and chains together:
`chids_segnames_list = [(item.getChid(), item.getSegname()) for item in atoms.getHierView()]`

Reply:
I looked at the code. To include the segment name in saveInteractionsPDB(), I have to have information about _freq_interactors, which are computed by calcFrequentInteractors using plip. But plip doesn't provide a segment name. I checked that. The user is forced to provide a structure with different chains or resnames. I was testing the code on the TriC/CCT structure (multiple chains). So, with plip we will not be able to distinguish segments anyway. If the chain name will be the same plip might count it as one molecule, and then even if we distinguish segments in saveInteractionsPDB() the number of _freq_interactors will be different compared to the number of atoms, and the program will crash.

Example output from plip:
[saltbridge(positive=pcharge(atoms=[<openbabel.pybel.Atom object at 0x7efe55e3c8e0>], atoms_orig_idx=[1219], type='positive', center=[-31.228, 46.733, 3.248], restype='LYS', resnr=161, reschain='A'), negative=lcharge(atoms=[<openbabel.pybel.Atom object at 0x7efe581bbb80>, <openbabel.pybel.Atom object at 0x7efe581c2bb0>, <openbabel.pybel.Atom object at 0x7efe581c13d0>, <openbabel.pybel.Atom object at 0x7efe581c1a30>, <openbabel.pybel.Atom object at 0x7efe581c1940>], orig_atoms=[<openbabel.pybel.Atom object at 0x7efe581c2e50>, <openbabel.pybel.Atom object at 0x7efe581c1850>, <openbabel.pybel.Atom object at 0x7efe581c1760>, <openbabel.pybel.Atom object at 0x7efe581c1670>, <openbabel.pybel.Atom object at 0x7efe581c1580>, <openbabel.pybel.Atom object at 0x7efe581c1490>], atoms_orig_idx=[12155, 12155, 12154, 12156, 12157, 12158], type='negative', center=(-27.99, 50.232, 5.798), fgroup='phosphate'), distance=5.406491006188768, protispos=True, resnr=161, restype='LYS', reschain='A', resnr_l=1, restype_l='ADP', reschain_l='A'),
 hbond(a=<openbabel.pybel.Atom object at 0x7efe581be1f0>, a_orig_idx=12150, d=<openbabel.pybel.Atom object at 0x7efe59318820>, d_orig_idx=3197, h=<openbabel.pybel.Atom object at 0x7efe585210d0>, distance_ah=2.56645614403395, distance_ad=3.181985072246572, angle=120.55688368507359, type='strong', protisdon=True, resnr=415, restype='ALA', reschain='A', resnr_l=1, restype_l='ADP', reschain_l='A', sidechain=False, atype='O3', dtype='Nam'),..]

Reply:
ok, then, let's leave it and the docs are clear enough anyway I think